iceoryx_posh/source/runtime/posh_runtime.cpp

```diff
@@ -507,10 +507,19 @@ PoshRuntime::findService(const capro::ServiceDescription& serviceDescription) no
 }
 
 
-void PoshRuntime::offerService(const capro::ServiceDescription& serviceDescription) noexcept
+bool PoshRuntime::offerService(const capro::ServiceDescription& serviceDescription) noexcept
 {
+    if (serviceDescription.getServiceIDString() == iox::capro::InvalidIDString
```

Comment:
I would propose to do the check not here because this is something which is somehow only related to the servicedescription.  We will also in other cases need a function which checks if the servicedescription is invalid.

Please create in service_description.hpp a function called `isvalid` which maybe return a bool. Inside of that you can check then against the invalid values. Check will be easier here because internally the two friends serviceid and instanceid are represented as u_int values which you can easy compare without converting.